6814 TOGGLE DAPI Refaktorering av steg 1 i årsavregningen

fredrik-oh opened the pull request:
https://jira.adeo.no/browse/MELOSYS-6814

api: https://github.com/navikt/melosys-api/pull/2519

Opprettet ny komponent tidligereGrunnlagsoversikt. 
Erstattet trygdeavgiftsPerioderTabell med inntektsperioderTabell.

thanglyphan committed:
6814 TOGGLE DAPI Refaktorering av steg 1 i årsavregningen

Opprettet ny komponent tidligereGrunnlagsoversikt.
Erstattet trygdeavgiftsPerioderTabell med inntektsperioderTabell.

frans-a commented on src/sider/aarsavregning/stegKomponenter/vurderingAarsavregning/komponenter/beregnetTrygdeavgiftDetaljer.tsx:
```diff
@@ -62,9 +62,9 @@ export const BeregnetTrygdeavgiftDetaljer = ({
 
   return (
-    <Nav.ExpansionCard aria-label="trygdeavgiftdetaljer">
+    <Nav.ExpansionCard aria-label="trygdeavgiftdetaljer" size="small">
```
Hvordan migrerer vi til Aksel-komponenter over tid?
Gir det f.eks. mening å ta i bruk https://aksel.nav.no/komponenter/core/expansioncard nå?

fredrik-oh replied:
Dette er den komponenten, vi har en samling "skallkomponenter" under navFrontend som overstyrer defaultprops (typisk size) og kan utvides til å ha egen styling ved behov.

frans-a commented on src/sider/aarsavregning/stegKomponenter/vurderingAarsavregning/komponenter/inntektsperioderTabell.tsx:
```diff
@@ -0,0 +1,57 @@
+              <Nav.Table.DataCell key={Utils._uuid()}>
+                {inntektsperiode.arbeidsgiversavgiftBetales ? "Ja" : "Nei"}
+              </Nav.Table.DataCell>
+              <Nav.Table.DataCell align="right" key={Utils._uuid()}>
```
`align="right"` ser mye mer til høyre enn skissen hos meg

fredrik-oh replied:
Jepp, følger table stylingen til aksel. Skal høre med Cathrin om hun vil ha det endret

frans-a commented on src/sider/aarsavregning/stegKomponenter/vurderingAarsavregning/komponenter/medlemskapsPerioderTabell.tsx:
```diff
@@ -12,18 +12,22 @@ const MedlemskapsPerioderTabell = ({ perioder }: { perioder?: Medlemskapsperiode
             </Nav.Table.DataCell>
-            <Nav.Table.DataCell key={Utils._uuid()}>
+            <Nav.Table.DataCell>
+              {KV.kodeTilTerm(medlemskapsPeriode.medlemskapstype, MKV.KTObjects.medlemskapstyper)}
```
Skal man vise "Frivillig medlemskap" eller bare "Frivillig" som i skissen?

fredrik-oh replied:
Skissen har oppdaterte tekster så jeg fikser

fredrik-oh replied:
Vent, dette er teksten fra kodeverk. Hvis det ønskes en endring i tekst der må det nok inn i akseptansekriteriene. Skal sjekke.

frans-a commented on src/sider/aarsavregning/stegKomponenter/vurderingAarsavregning/vurderingAarsavregning.tsx:
```diff
@@ -325,6 +324,7 @@ export const VurderingAarsavregning = ({ bekreft, oppdaterStatus }: Props) => {
   useEffect(() => {
     oppdaterStatus(stegErGyldig);
   }, [stegErGyldig]);
+  console.log(lagretTrygdeavgift?.tidligereGrunnlagsopplysninger);
```
Hva gjør man med disse og 
>Unexpected console statement

?

fredrik-oh replied:
Lint logger det som warning. Skal i utgangspunktet ikke være med

frans-a commented on src/sider/aarsavregning/stegKomponenter/vurderingAarsavregning/vurderingAarsavregning.tsx:
```diff
@@ -354,11 +354,11 @@ export const VurderingAarsavregning = ({ bekreft, oppdaterStatus }: Props) => {
+            inntektsperioder={lagretTrygdeavgift.tidligereGrunnlagsopplysninger.trygdeavgiftsgrunnlag.inntektskperioder}
             avgift={lagretTrygdeavgift.tidligereGrunnlagsopplysninger.avgift}
           />
         </>
```
Ser bra ut med ny komponent

Commit:
qa justeringer

overstyrer kodeverksterm for medlemskapstype i medlemskapsPerioderTabell.tsx

fjernet console.log
